design/one-pager-resource-usages.md

```diff
@@ -0,0 +1,346 @@
+  would be more appropriate.
+* Establish foundations for modelling the consumption of a managed resource in
+  order to eventually ensure network level connectivity between.
+
+Note that while this document may lay the foundation for network level
+connectivity constructs, it defers discussing that problem space in detail.
+
+## Proposed Design
+
+This document proposes:
+
+* That the existing resource class and managed resource kinds become cluster
+  scoped, rather than namespaced.
+* The introduction of cluster scoped, 'resource secrets' aligned with existing
+  managed resource kinds, for example `RDSInstanceSecret`, or
+  `CloudMemorystoreInstanceSecret`.
+* The introduction of namespaced 'resource usages' aligned with existing
+  resource claim kinds, for example `PostgreInstanceSQLUsage`, or
+  `RedisClusterUsage`.
+
+### Resource Secrets
+
+A resource secret is identical in purpose to the standard Kubernetes `Secret`,
+but cluster scoped and with a schema communicating the intent of its associated
+managed resource. Resource secrets store _only_ sensitive connection data. For
+example:
+
+```yaml
+apiVersion: database.aws.crossplane.io/v1alpha1
+kind: RDSInstanceSecret
+metadata:
+  name: rds-some-uuid
+# Values are base64 encoded, presented decoded here for readability.
+# Note the lack of non-sensitive data such as username, URI, etc.
+masterPassword: crossplane_generated_password
+```
+
+Cluster scoped resource secrets enable managed resources and their classes to be
+cluster scoped, obviating the (arguably) awkward `crossplane-system` namespace.
+Resource secrets cannot easily be consumed by a `Pod` as there is no mechanism
+with which to expose them as files or environment variables inside a container.
+This is a desirable property; it requires a resource claim and usage (which may
+expose connection details in a consumable fashion) to exist in order to consume
+an external resource. No controller is necessary to reconcile resource secrets;
+they exist simply as a slight obfuscation of sensitive connection details such
+as passwords that might otherwise be stored directly as part of their managed
+resource's `.status`.
+
+### Resource Usages
+
+A resource usage builds on the resource claim concept, specifying how a claim's
+underlying managed resource's connection data should be exposed to a specific
+consumer, as well as 'solving' the usage, for example by ensuring the consumer
+has network connectivity to the managed resource, or provisioning usage scoped
+credentials with access to the managed resource. Resource usages, like resource
+claims, are namespaced. They must exist in the same namespace as their claim and
+consumer.
+
+```yaml
+apiVersion: storage.crossplane.io/v1alpha1
+kind: PostgreSQLInstanceUsage
+metadata:
+  namespace: cool-team
+  name: cool-sock-product-db
+spec:
+  claimRef:
+    apiVersion: storage.crossplane.io/v1alpha1
+    kind: PostgreSQLInstance
+    namespace: cool-team
+    name: cool-sock-db
```

Comment:
We should ignore namespace here. We should not support using claims in other namespaces.